src/main/java/com/github/hcsp/encapsulation/Cat.java

```diff
@@ -5,20 +5,32 @@
     private String name;
     private int age;
 
+    public Cat(String name, int age) {
```

Comment:
```suggestion
    private Cat(String name, int age) {
```

有了工厂方法，构造器就可以`private`了

Reply:
刚想到